opengemini-client-api/src/main/java/io/opengemini/client/api/Point.java

```diff
@@ -2,53 +2,131 @@
+            return validFields;
+        }
 
-        sb.append(" ");
         boolean firstField = true;
         for (Map.Entry<String, Object> entry : fields.entrySet()) {
+            Object fieldValue = entry.getValue();
+            if (fieldValue == null || isNotFinite(fieldValue)) {
+                continue;
+            }
+
             if (firstField) {
                 firstField = false;
             } else {
-                sb.append(",");
+                sb.append(',');
             }
 
-            sb.append(entry.getKey()).append("=");
-            if (entry.getValue() instanceof String) {
-                sb.append("\"").append(entry.getValue()).append("\"");
-            } else if ((entry.getValue() instanceof Integer) || (entry.getValue() instanceof Long)) {
-                sb.append(entry.getValue()).append("i");
+            String fieldKey = entry.getKey();
+            appendWithEscape(sb, fieldKey, FIELD_KEY_ESCAPE_CHARACTERS);
+            sb.append('=');
+            if (fieldValue instanceof Number) {
+                if (fieldValue instanceof Double || fieldValue instanceof Float || fieldValue instanceof BigDecimal) {
+                    sb.append(fieldValue);
+                } else {
+                    sb.append(fieldValue).append('i');
+                }
+            } else if (fieldValue instanceof String) {
+                String stringValue = (String) fieldValue;
+                sb.append('"');
+                appendWithEscape(sb, stringValue, FIELD_VALUE_ESCAPE_CHARACTERS);
+                sb.append('"');
             } else {
-                throw new Exception("ss");
+                sb.append(fieldValue);
             }
+            validFields++;
         }
-        sb.append(" ");
-        sb.append(converteTime());
-        return sb.toString();
+
+        return validFields;
     }
 
-    private String converteTime() {
-        if (time == 0) {
-            return "";
+    private void appendTimestamp(StringBuilder sb) {
+        sb.append(' ');
+        if (time != 0 && precision != null) {
+            sb.append(precision.getTimeUnit().toNanos(time));
+        }
+    }
+
+    @Override
+    public String toString() {
+        return lineProtocol();
+    }
+
+    private static void appendWithEscape(StringBuilder sb, String origin, char[] escapeChars) {
+        for (char c : origin.toCharArray()) {
+            if (shouldEscape(c, escapeChars)) {
+                sb.append("\\");
+            }
+            sb.append(c);
         }
-        return String.valueOf(time);
+    }
+
+    private static boolean shouldEscape(char c, char[] escapeChars) {
+        for (char escapeChar : escapeChars) {
+            if (escapeChar == c) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isNotFinite(final Object value) {
+        return value instanceof Double && !Double.isFinite((Double) value)
```

Comment:
Although the precedence of logical operators && and || in Java has been clearly defined, with && higher than ||, explicit grouping can make the code easier to read and understand. Alternatively, some sub-functions can be used to enhance logical grouping through function naming.